### PR TITLE
Completed Iteration 2.

### PR DESCRIPTION
**Pull request by evgenii-komornyi**
### Iteration 2 Requirements

- [x] 1. Add health metric for each animal
- [x] 2. Decrease 0.5 health on each move
- [x] 3. Increase health when antelope is eaten
- [x] 4. Add die function for lack of health
- [x] 5. Birth function when 2 animals are near each other for 3 consecutive rounds

### CODE REVIEW

File: GameEngine/Entities/Antilope.cs [hunk 1]
```diff
@@ -81,11 +82,12 @@ public override void MoveAnimal(Board board, Animal animal, List<Animal> animals
...
-                pointsCoordinates.FirstXPoint = lion.CoordinateX;
-                pointsCoordinates.SecondXPoint = lion.CoordinateX;
-                pointsCoordinates.FirstYPoint = lion.CoordinateY;
-                pointsCoordinates.SecondYPoint = lion.CoordinateY;
+                PointsCoordinates pointsCoordinates = new PointsCoordinates
+                {
+                    FirstXPoint = lion.CoordinateX,
+                    SecondXPoint = lion.CoordinateX,
+                    FirstYPoint = lion.CoordinateY,
+                    SecondYPoint = lion.CoordinateY            };
```
Comment (ArvisZalitis) [inline GameEngine/Entities/Antilope.cs]: Small alignment issues :D

Comment (ArvisZalitis) [reply GameEngine/Entities/Antilope.cs]: image

File: GameEngine/Entities/Lion.cs [hunk 2]
```diff
@@ -143,11 +148,13 @@ private int CalculateMinDistanceToAntilopeByFreeCells(List<NewAnimalCoordinates>
...
-                pointsCoordinates.SecondXPoint = antilope.CoordinateX;
-                pointsCoordinates.FirstYPoint = cell.NewYCoordinate;
-                pointsCoordinates.SecondYPoint = antilope.CoordinateY;
+                PointsCoordinates pointsCoordinates = new PointsCoordinates
+                {
+                    FirstXPoint = cell.NewXCoordinate,
+                    SecondXPoint = antilope.CoordinateX,
+                    FirstYPoint = cell.NewYCoordinate,
+                    SecondYPoint = antilope.CoordinateY
+            };
```
Comment (ArvisZalitis) [inline GameEngine/Entities/Lion.cs]: Same comment about }; allayment.

**Review by ArvisZalitis — APPROVED** [review]
Look good, some small fixes needed but nothing big.

File: GameEngine/Entities/Antilope.cs (hunk 1 above)
Comment (evgenii-komornyi) [reply GameEngine/Entities/Antilope.cs]: Done.

File: GameEngine/Entities/Lion.cs (hunk 2 above)
Comment (evgenii-komornyi) [reply GameEngine/Entities/Lion.cs]: Done.

**Review by EdgarsBondars — APPROVED** [review]

**Pull request merged**